### ISSUE TITLE
Update dependencies + fix issues related with that

### ISSUE THREAD

**Issue by kmdrGroch** [stryker-mutator/stryker#1589]
By mistake, I have updated some of the files (it was typescript), and after some bug fixing related to that, I have noticed that you are missing a lot of @types and most of them are obsolete!
for example:
currently installed
```
"@types/glob": "^7.1.0",
"@types/istanbul": "^0.4.29",
"@types/karma": "^3.0.0",
"@types/lodash": "^4.14.110",
"@types/mkdirp": "^0.5.2",
"@types/mocha": "^2.2.44",
"@types/node": "^10.11.5",
"@types/rimraf": "2.0.2",
"@types/sinon": "5.0.5",
"@types/sinon-chai": "^3.2.0"
```

current:
```
"@types/glob": "^7.1.1",
"@types/istanbul": "^0.4.30",
"@types/karma": "^3.0.3",
"@types/lodash": "^4.14.134",
"@types/mkdirp": "^0.5.2",
"@types/mocha": "^5.2.7",
"@types/node": "^10.11.5",
"@types/rimraf": "2.0.2",
"@types/sinon": "7.0.13",
"@types/sinon-chai": "^3.2.2",
```
some of them don't have super changes but other like sinon / mocha are quite obsolete
\+ you in one package i have noticed you don't use @types for jest (it can cause some type error in the future)

**Comment by nicojs** [comment]
With regards to TypeScript 3.5, I'm waiting for the fix for https://github.com/microsoft/TypeScript/issues/31676 to be released (will hopefully be 3.5.2). After that I we will be able to update TypeScript itself.

TL;DR;
I'm happy to accept a PR that changes all `^` (minor version) ranges to `~` (patch version) ranges. For more info on `^` and `~` see https://docs.npmjs.com/about-semantic-versioning#using-semantic-versioning-to-specify-update-types-your-package-can-accept. 
I'm also happy to accept a PR for mocha types.

Long version:

With regards to the `@types` dependencies, they're not exactly "obsolete". We're using [dependabot](https://dependabot.com/) to automagically update our dependencies. The thing is, it will only create a PR if the new version is outside of the semver range. For example `"@types/lodash": "^4.14.134",` is still in ranche of `"@types/lodash": "^4.14.110",` and so it will not create a PR for it. We're also happy with not following patch releases as it would create a lot of PR's. 

I would agree that we should never use `^`, instead use `~`. That way we will get a PR for minor releases.

The only one that is really outdated are mocha and sinon. Updating sinon will be a lot fo work because of this issue: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/36019. I'm not sure why mocha isn't updated.

**Comment by kmdrGroch** [comment]
@nicojs what about mocha, we are using version  "@types/mocha": "^2.2.44", where the newest one is "^5.2.7",
